cli/flox/doc-catalog/manifest.toml.md

```diff
@@ -0,0 +1,366 @@
+    These directories from all packages in the
+    environment are merged when building the environment.
+    Two packages that provide the same `/bin/foo` file cause a conflict,
+    and it's ambiguous which file should ultimately be placed into the
+    environment.
+    Such conflicts can be resolved by assigning different priorities
+    to the conflicting packages.
+
+    The default priority is 5.
+    Packages with a lower `priority` value will take precedence over packages
+    with higher `priority` values.
+
+## `[vars]`
+
+The `[vars]` section allows you to define environment variables for your
+environment that are set during environment activation.
+The environment variables specified here cannot reference one another.
+The names and values of the environment variables are copied verbatim into the
+activation script,
+so capitalization will be preserved.
+
+Example:
+```toml
+[vars]
+DB_URL = "http://localhost:2000"
+SERVER_PORT = "3000"
+```
+
+## `[hook]`
+
+The `on-activate` script in the `[hook]` section is useful for performing
+initialization in a predictable Bash shell environment.
+
+### `on-activate`
+
+The `on-activate` script is sourced from a **bash** shell,
+and it can be useful for spawning processes, dynamically setting environment
+variables, and creating files and directories to be used by the subsequent
+profile scripts, commands, and shells.
+
+Hook scripts inherit environment variables set in the `[vars]` section,
+and variables set here will in turn be inherited by the `[profile]` scripts
+described below.
+
+Any output written to `stdout` in a hook script is redirected to `stderr` to
+avoid it being mixed with the output of profile section scripts that write to
+`stdout` for "in-place" activations.
+
+```toml
+[hook]
+on-activate = """
+    # Interact with the tty as you would in any script
+    echo "Starting up $FLOX_ENV_DESCRIPTION environment ..."
+    read -e -p "Favourite colour or favorite color? " value
+
+    # Set variables, create files and directories
+    venv_dir="$(mktemp -d)"
+    export venv_dir
+
+    # Perform initialization steps, e.g. create a python venv
+    python -m venv "$venv_dir"
+
+    # Invoke apps that configure the environment via stdout
+    eval "$(ssh-agent)"
+"""
+```
+
+The `on-activate` script is not re-run when activations are nested.
+A nested activation can occur when an environment is already active and either
+`eval "$(flox activate)"` or `flox activate -- CMD` is run.
+In this scenario, `on-activate` is not re-run.
+Currently, environment variables set by the first run of the `on-activate`
+script are captured and then later set by the nested activation,
+but this behavior may change.
+
+It is best to write hooks defensively, assuming the user is using the
+environment from any directory on their machine.
+
+### `script` - DEPRECATED
+This field was deprecated in favor of the `profile` section.
+It will be removed in a later release.
```

Comment:
```suggestion
The `hook.script` field will be removed in a later release.
```
I know this is copy-paste but there's a bit of ambiguity in how it was originally written.

Reply:
I added that wording to `doc/manifest.toml.md`. I completely dropped this sentence from the catalog version because we don't support `hook.script` at all with the catalog code